tests/testthat/test-app.R

```diff
@@ -249,7 +251,13 @@ test_that("Setting options in various places works", {
 
   # Calls to options(shiny.port = xxx) within app.R should also work reliably
   expect_port(runApp(file.path(appDir, "option.R")), 7777)
-  # But still overrideable
+  # Ensure that option was unset
+  expect_null(getOption("shiny.port"))
```

Comment:
This test makes the assumption that the starting value for `shiny.port` was NULL, but the app itself restores whatever the starting value was, even if it was something other than NULL.

For these tests, maybe set `shiny.port` to NULL before doing them, or save the original value for `shiny.port`, and compare to that in the tests?